src/rules.jl

```diff
@@ -135,6 +136,47 @@ function Base.show(io::IO, o::RMSProp)
     print(io, "; centred = ", o.centred, ")")
 end
 
+
+"""
+    Rprop(η = 1f-3, ℓ = (5f-1, 1.2f0), Γ = (1f-6, 50f0))
+
+Optimizer using the
+[Rprop](https://ieeexplore.ieee.org/document/298623) algorithm. An full-batch
+learning algorithm that depends only on the sign of the gradient.
+
+# Parameters
+- Learning rate (`η`): Amount by which gradients are discounted before updating
+                       the weights.
+                      
+- Scaling factors (`ℓ::Tuple`): Multiplicative increase and decrease factors.
+
+- Step sizes (`Γ::Tuple`): Mminimal and maximal allowed step sizes.
+"""
+struct Rprop{T} <: AbstractRule
+    eta::T
+    ell::Tuple{T,T}
+    gamma::Tuple{T,T}
+end
+
+Rprop(η = 1f-3, ℓ = (5f-1, 1.2f0), Γ = (1f-6, 50f0)) = Rprop{typeof(η)}(η, ℓ, Γ)
+
+init(o::Rprop, x::AbstractArray) = (zero(x), onevalue(o.eta, x))
+
+function apply!(o::Rprop, state, x, dx)
+    ℓ, Γ = o.ell, o.gamma
+    g, η = state
+
+    signs = map(s-> s > 0 ? ℓ[2] :
+                    s < 0 ? ℓ[1] : one(eltype(dx)),
+                sign.(g .* dx))
```

Comment:
```suggestion
    signs = map(g, dx) do g, dx
        s = sign(g * dx)
        s > 0 ? ℓ[2] : s < 0 ? ℓ[1] : one(eltype(dx))
    end
```
Manually fusing these seems about 2x faster for me on CPU and GPU, while saving on an intermediate allocation from `sign.(g .* dx)`.